src/serlo-editor/plugins/text/plugins/with-empty-lines-restriction.ts

```diff
@@ -0,0 +1,103 @@
+import { Editor, Element, Node, Path, Transforms } from 'slate'
+import { ReactEditor } from 'slate-react'
+
+// Slate normalization docs: https://docs.slatejs.org/concepts/11-normalizing
+// While editing a Text plugin: Two adjacent empty lines are allowed,
+// so that one of them could be transformed into another plugin,
+// using the plus button.
+// On Text plugin blur: Only one empty line is allowed.
+export const withEmptyLinesRestriction = (editor: Editor) => {
+  const { normalizeNode } = editor
+
+  editor.normalizeNode = (entry, { operation } = { operation: undefined }) => {
+    const [node, path] = entry
+    const isBlur = (operation as unknown as string) === 'blur'
+
+    if (shouldNodeBeRemoved(editor, node, path, isBlur)) {
+      Transforms.removeNodes(editor, { at: path })
+      // TODO: Show warning?
+      if (isBlur) ReactEditor.blur(editor)
```

Comment:
Is this line needed? As far as I understand, the blur event already occurred at this point.